cmd/agent/main.go

```diff
@@ -94,6 +119,36 @@ func compressGzip(data []byte) ([]byte, error) {
 	return b.Bytes(), nil
 }
 
+func getHash(data []byte, strkey string) (string, error) {
+	h := hmac.New(sha256.New, []byte(strkey))
+	_, err := h.Write(data)
+	if err != nil {
+		return "", err
+	}
+
+	sign := h.Sum(nil)
+	return hex.EncodeToString(sign), nil
+}
+
+// source - канал со снятыми метриками, содержит весь пакет
+// n - на сколько запросов/работников можно разделить пакет метрик
+func SplitBatch(source <-chan api.Metrics, n int) []<-chan api.Metrics {
+	dests := make([]<-chan api.Metrics, 0) // Создать срез dests
```

Comment:
Так как знаем размер `n` можно сразу выделить необходимый объем памяти.
```suggestion
	dests := make([]<-chan api.Metrics, 0, n) // Создать срез dests
```

Reply:
И правда. Тогда это будет уже массив, а не срез, и там нужно тогда не `append`, а по индексу его заполнять.

Reply:
Изменил тут [slice -> allocated array](https://github.com/xoxloviwan/go-monitor/pull/14/commits/4790b9ce8e9045415eac38985ca84f9e04d68217)